core/src/components/segment-button/segment-button.tsx

```diff
@@ -67,7 +67,51 @@ export class SegmentButton implements ComponentInterface, ButtonInterface {
     this.updateState();
   }
 
-  connectedCallback() {
+  private getNextSiblingOfType<T extends Element>(element: Element): T | null {
+    let sibling = element.nextSibling;
+    while (sibling) {
+      if (sibling.nodeType === Node.ELEMENT_NODE && (sibling as T) !== null) {
+        return sibling as T;
+      }
+      sibling = sibling.nextSibling;
+    }
+    return null;
+  }
+
+  private waitForSegmentContent(ionSegment: HTMLIonSegmentElement | null, contentId: string): Promise<HTMLElement> {
+    return new Promise((resolve, reject) => {
+      if (!ionSegment) {
+        reject(new Error(`Segment not found when looking for Segment Content`));
+      }
+
+      let timeoutId: any = null;
```

Comment:
let's properly type this
```suggestion
      let timeoutId: number | undefined = undefined;
```